drivers/i3c/Kconfig.stm32

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2024 Your Company
+#
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_STM32_I3C := st,stm32-i3c
+module = I3C_STM32
+module-str = i3c_stm32
+
+source "subsys/logging/Kconfig.template.log_config"
+config I3C_STM32
+		bool "STM32 I3C driver support"
+		depends on DT_HAS_ST_STM32_I3C_ENABLED
+		select USE_STM32_HAL_I3C
+		default y
+		help
+			Enable support for I3C on STM32 microcontrollers.
+
+config I3C_STM32_POLL
+		bool "I3C polling mode for STM32"
+		depends on I3C_STM32
+		default n
+		help
+			Enables polling mode for I3C on STM32 microcontrollers.
```

Comment:
Question to maintainers (@dcpleung, @XenuIsWatching)
Is there a real interest in having this POLL option ?

If it's disabled by default, and doesn't provide a significant interest, it might be better to remove it right now to simplify driver code and long term maintenance.
We can find poll based code in few other drivers, but that's mostly for historical reasons as in general they are not used and present just because they where the initial method implemented.

@ExaltZephyr Any reason you'd like to keep this `I3C_STM32_POLL`option ?

Reply:
Unless there is a legitimate use case, I recommend removing it.